thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/views/TimeSeriesCompareView.js

```diff
@@ -17,41 +17,57 @@ function TimeSeriesCompareView(timeSeriesCompareModel) {
   this.contributor_table_placeHolderId = "#contributor-table-placeholder";
   this.heatmapRenderEvent = new Event(this);
 
-  this.viewParams;
+  this.viewParams = {};
+  this.chart;
 }
 
 TimeSeriesCompareView.prototype = {
-  render: function () {
-    var self = this;
+  render() {
     if (this.timeSeriesCompareModel.subDimensionContributionDetails) {
+      const heatmapCurrentStart = this.timeSeriesCompareModel.heatMapCurrentStart;
+      const heatmapCurrentEnd = this.timeSeriesCompareModel.heatMapCurrentEnd;
+
       this.renderChartSection();
+      if (heatmapCurrentStart && heatmapCurrentEnd) {
+        this.loadAnomalyRegion(heatmapCurrentStart, heatmapCurrentEnd);
+      }
       this.renderPercentageChangeSection();
       this.setupListenersForDetailsAndCumulativeCheckBoxes();
 
       // set initial view params for rendering heatmap
       this.viewParams = {
-        metricId: self.timeSeriesCompareModel.metricId,
-        currentStart: self.timeSeriesCompareModel.currentStart,
-        currentEnd: self.timeSeriesCompareModel.currentEnd,
-        baselineStart: self.timeSeriesCompareModel.baselineStart,
-        baselineEnd: self.timeSeriesCompareModel.baselineEnd,
-        heatmapFilters: self.timeSeriesCompareModel.filters
+        metricId: this.timeSeriesCompareModel.metricId,
+        heatmapCurrentStart: this.timeSeriesCompareModel.heatmapCurrentStart || this.timeSeriesCompareModel.currentStart,
+        heatmapCurrentEnd: this.timeSeriesCompareModel.heatMapCurrentEnd || this.timeSeriesCompareModel.currentEnd,
+        heatmapBaselineStart: this.timeSeriesCompareModel.baselineStart,
+        heatmapBaselineEnd: this.timeSeriesCompareModel.baselineEnd,
+        heatmapFilters: Object.assign({}, this.timeSeriesCompareModel.filters)
       };
     }
   },
 
+  destroy() {
+    $('#timeseries-contributor-placeholder').children().remove();
```

Comment:
Lines 50 - 51: I believe you can target multiple selectors at one go, e.g. `$('#timeseries-contributor-placeholder, #contributor-table-placeholder').children().remove();`